nsq

fw42 opened the pull request:
Adds a nsq worker.

We've been using something very similar to this in production for a while and I thought it might be useful to contribute it back upstream so that others can benefit too.

fw42 commented:
Thanks for the quick response. I'll take a look at that shortly. Just wanted to confirm first whether, in principle, there's even any interest in having this merged before I invest more time into it. But sounds like that's the case?

dmachard commented:
Yes, absolutely!  Your contribution would be valuable — just follow the guidelines and add the tests/docs when you can. Also, feel free to give the project a star ⭐ or share it if you find it useful!

fw42 commented:
Cleaned this up a little bit and added some docs and tests as you requested.

The test feels a bit awkward with the mock producer. Suggestions welcome how to improve this. I tried spawning a real TCP server at first but it turned ugly quickly trying to emulate the nsq protocol.

fw42 commented:
FYI before this is ready to get merged, I would like to add support for [ConnectToNSQLookupd](https://pkg.go.dev/github.com/nsqio/go-nsq#Consumer.ConnectToNSQLookupd), but that should be a minor change and I would already appreciate some early feedback on the current state.

fw42 commented:
Hmm I see that the tests failed on CI. They are passing locally, so not sure what's going on. Will investigate.

fw42 commented:
Ok both the linting error and the race condition should be fixed now. Sorry for the noise.

fw42 commented:
Pushed one more change to simplify the tests a bit. I think this should be ready for review now.

dmachard commented:
Look good to me
